core/src/ru/spbau/blackout/network/AndroidClientThread.java

```diff
@@ -0,0 +1,92 @@
+package ru.spbau.blackout.network;
+
+import com.badlogic.gdx.Gdx;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.Socket;
+import java.net.UnknownHostException;
+
+import ru.spbau.blackout.BlackoutGame;
+import ru.spbau.blackout.screens.MenuScreen;
+import ru.spbau.blackout.screens.MultiplayerTable;
+import ru.spbau.blackout.screens.PlayScreenTable;
+
+import static ru.spbau.blackout.BlackoutGame.HOST_NAME;
+import static ru.spbau.blackout.BlackoutGame.PORT_NUMBER;
+
+public class AndroidClientThread extends Thread {
```

Comment:
But I need a way of finishing this thread earlier with something like interrupting it (because of Back button). I can use a flag there, but isn't the runnable with the property of interrupting it is more like Thread semantically?